src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt

```diff
@@ -0,0 +1,123 @@
+        rollupId = rollup.id
+        logger.info("Attempting to create a rollup job $rollupId for index $indexName")
+
+        val indexRollupRequest = IndexRollupRequest(rollup, WriteRequest.RefreshPolicy.IMMEDIATE)
+
+        try {
+            val response = withContext(Dispatchers.IO) { client.execute(IndexRollupAction.INSTANCE, indexRollupRequest).actionGet() }
+            logger.info("Received status ${response.status.status} on trying to create rollup job $rollupId")
+
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to getSuccessMessage(rollupId!!, indexName))
+        } catch (e: VersionConflictEngineException) {
+            val message = getFailedJobExistsMessage(rollupId!!, indexName)
+            logger.info(message)
+            if (rollupId == previousRunRollupId && hasPreviousRollupAttemptFailed != null && hasPreviousRollupAttemptFailed!!) {
+                withContext(Dispatchers.IO) {
+                    startRollupJob()
+                }
+            } else {
+                stepStatus = StepStatus.COMPLETED
+                info = mapOf("info" to message)
+            }
+        } catch (e: Exception) {
+            val message = getFailedMessage(rollupId!!, indexName)
+            logger.error(message, e)
+            stepStatus = StepStatus.FAILED
+            info = mapOf("message" to message, "cause" to "${e.message}")
+        }
+
+        return this
+    }
+
+    override fun getUpdatedManagedIndexMetaData(currentMetaData: ManagedIndexMetaData): ManagedIndexMetaData {
+        val currentActionMetaData = currentMetaData.actionMetaData
+        return currentMetaData.copy(
+                actionMetaData = currentActionMetaData?.copy(actionProperties = ActionProperties(rollupId = rollupId)),
+                stepMetaData = StepMetaData(name, getStepStartTime().toEpochMilli(), stepStatus),
+                transitionTo = null,
+                info = info
+        )
+    }
+
+    private fun startRollupJob() {
+        logger.info("Attempting to re-start the job $rollupId")
+        try {
+            val startRollupRequest = StartRollupRequest(rollupId!!)
+            client.execute(StartRollupAction.INSTANCE, startRollupRequest).actionGet()
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to getSuccessRestartMessage(rollupId!!))
+        } catch (e: Exception) {
+            val message = getFailedToStartMessage(rollupId!!)
+            logger.error(message, e)
+            stepStatus = StepStatus.CONDITION_NOT_MET
+            info = mapOf("message" to message)
+        }
+    }
+
+    companion object {
+        const val name = "attempt_create_rollup"
+        fun getFailedMessage(rollupId: String, index: String) = "Failed to create the rollup job [$rollupId] for index [$index]"
+        fun getFailedJobExistsMessage(rollupId: String, index: String) = "Rollup job [$rollupId] already exists for index [$index]"
```

Comment:
This is the message being shown to user, in the flow above it seems like we show this even when setting to COMPLETED which means it'll move on to the next step. So can we change it to reflect what is going to happen, like Rollup job [$rollupId] already exists, skipping creation [index=$index]. Or something better that you can come up with.

Reply:
makes sense. changed